src/contracts/BPool.sol

```diff
@@ -71,7 +71,7 @@ contract BPool is BToken, BMath, IBPool {
   }
 
   /// @inheritdoc IBPool
-  function finalize() external _logs_ _lock_ {
+  function finalize() public virtual _logs_ _lock_ {
```

Comment:
do we need it `public`? does `super.finalize` not work with external methods?

Reply:
hmmm ok, it does, given that we need to make it `public`, and we're actually risking the fact that some contract above it doesn't implement it, how about the `_afterFinalize` virtual internal?